### PR TITLE
CORE-4804 - Improve package structure of link manager module

### PR DESCRIPTION
**Pull request by dimosr**
Creating a cleaner package structure for the link manager module. There are no code changes here, only from moving classes into packages, except a few functions that are not `internal` anymore since the corresponding class is.

### CODE REVIEW

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/common/PublicKeyReader.kt [hunk 1]
```diff
@@ -10,7 +10,7 @@ import java.security.PublicKey
 
 internal class PublicKeyReader {
     companion object {
-        internal fun PublicKey.toKeyAlgorithm(): KeyAlgorithm {
```
Comment (williamvigorr3) [inline components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/common/PublicKeyReader.kt:13]: Can't these stay as `internal` as they are still only used in this module?

**Review by yift-r3 — DISMISSED** [review]

**Review by williamvigorr3 — COMMENTED** [review]
I have manually checked everything got moved to where we agreed on slack.

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/common/PublicKeyReader.kt (hunk 1 above)
Comment (yift-r3) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/common/PublicKeyReader.kt:13]: (The class is internal)

Comment (williamvigorr3) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/common/PublicKeyReader.kt:13]: Probably doesn't matter then.

**Review by williamvigorr3 — DISMISSED** [review]
LGTM

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/common/PublicKeyReader.kt (hunk 1 above)
Comment (dimosr) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/common/PublicKeyReader.kt:13]: It's a valid point you're making. It was IntelliJ that complained at some point for it, but unfortunately I didn't keep track of the warning message. So, I can't recall what was it and can't seem to reproduce it locally. Let me give it a try and remove them.

Comment (dimosr) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/common/PublicKeyReader.kt:13]: sorry I just saw Yiftach's comment. That makes me even more confused as to why Intellij complained then.

Let me revert the change as well. I suppose if the class is `internal`, it doesn't make sense to also have the methods `internal`.

Comment (williamvigorr3) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/common/PublicKeyReader.kt:13]: I noticed Intelij sometimes gets confused if you use something internal in a test (possibly only in an integration test). It underlines it in red but the test will still run.

**Review by yift-r3 — APPROVED** [review]

**Review by williamvigorr3 — APPROVED** [review]
LGTM

**Pull request merged**